Keyboard cursor customization

C-Loftus opened the pull request:
Addresses https://github.com/cursorless-dev/cursorless/issues/2401, the ability to have custom cursor styling exclusively when in cursorless keyboard mode.

Tried to follow the repo style as best as possible but please let me know if another way besides a record mapper is more idiomatic for TS.

## Checklist

- [/] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [x] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
    - _I could add docs in the visual a11y section but unclear if experimental keyboard support belongs there_
- [x] I have not broken the cheatsheet

## Example 
An image showing the cursor start out with a custom type, then change into `block` when keyboard mode is activated, thus adding more contrast.

![](https://github.com/cursorless-dev/cursorless/assets/70598503/b0dfd7bc-daa0-40bc-b052-20aed31c5d6e)

pokey reviewed (commented):
This is a good start, but won't update the cursor until the next restart, I believe. We pretty much always reflect changes to settings in real time, so eg if you are in cursorless keyboard modal mode and change this setting, the cursor should update immediately. It's a little bit of work to get that to happen, so lmk if you need help and we can discuss

C-Loftus commented:
- Fixed this up a bit. Reads every time `modeOn` is called. Works for me locally.
    - I tested to be sure again, and you are right Pokey, this did not update previously and had the same cursor style for an entire session even if it was changed 
- Was not sure on watching the setting and updating in real time. Probably not needed since keyboard mode would be toggled on/off very frequently. 
    - As discussed in the meetup this works fine at least for me and it doesn't need to touch other parts of the codebase.  
 
- Changed the setting names so they match the same formatting as the names within the standard vscode cursor style setting.
    -  I had to disable an eslint setting to allow for kebab'd property names but I think this is better than the alternative of having a different name.

pokey commented:
> * Fixed this up a bit. Reads every time `modeOn` is called. Works for me locally.

👍

>       * I tested to be sure again, and you are right Pokey, this did not update previously and had the same cursor style for an entire session even if it was changed

👍

>     * Was not sure on watching the setting and updating in real time. Probably not needed since keyboard mode would be toggled on/off very frequently.

Yeah fine for now I think

>       * As discussed in the meetup this works fine at least for me and it doesn't need to touch other parts of the codebase.

👍

>     * Changed the setting names so they match the same formatting as the names within the standard vscode cursor style setting.

We always use camel case, but I think you're right to mirror VSCode ones

>       * I had to disable an eslint setting to allow for kebab'd property names but I think this is better than the alternative of having a different name.

See my workaround
